apps/shared/avifpng.c

```diff
@@ -17,6 +18,120 @@ typedef png_bytep png_iccp_datap;
 typedef png_charp png_iccp_datap;
 #endif
 
+static avifBool avifIsHexDigit(char hexDigit, uint8_t * decimalValue)
+{
+    if ((hexDigit >= 'A') && (hexDigit <= 'F')) {
+        *decimalValue = (uint8_t)(hexDigit - 'A') + 10;
```

Comment:
Nit: The `(uint8_t)` cast needs to be applied to the entire expression, otherwise the result of the cast will be automatically converted to `int` before the addition with 10.

```
        *decimalValue = (uint8_t)(hexDigit - 'A' + 10);
```

Alternatively, cast only if the compiler warns.